extensions/http/src/jolie/net/HttpProtocol.java

```diff
@@ -936,7 +941,11 @@ private void parseXML( HttpMessage message, Value value, String charset )
 	private static void parseJson( HttpMessage message, Value value, boolean strictEncoding, String charset )
 		throws IOException
 	{
-		JsUtils.parseJsonIntoValue( new InputStreamReader( new ByteArrayInputStream( message.content() ), charset ), value, strictEncoding );
+		try {
+			JsUtils.parseJsonIntoValue( new InputStreamReader( new ByteArrayInputStream( message.content() ), charset ), value, strictEncoding );
```

Comment:
Shouldn't this throw a fault? This way we get a string message and maybe the programmer was not expecting it in the type, hence raising an unexpected TypeMismatch fault.